library/src/main/java/com/novoda/downloadmanager/lib/DownloadProvider.java

```diff
@@ -408,10 +408,12 @@ private Uri insertDownload(Uri uri, ContentValues values, SQLiteDatabase db, int
                 dest = DownloadsDestination.DESTINATION_CACHE_PARTITION;
             }
             if (dest == DownloadsDestination.DESTINATION_FILE_URI) {
-                getContext().enforcePermission(
-                        android.Manifest.permission.WRITE_EXTERNAL_STORAGE, Binder.getCallingPid(), Binder.getCallingUid(),
-                        "need WRITE_EXTERNAL_STORAGE permission to use DESTINATION_FILE_URI");
-                checkFileUriDestination(values);
+                if (!getFileUriDestinationPath(values).contains("/" + getContext().getPackageName() + "/")) {
```

Comment:
No problem! Perhaps you do some file path checking against [`Environment#getDataDirectory()`](http://developer.android.com/reference/android/os/Environment.html#getDataDirectory()) instead?